client/lib/create-selector/index.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import memoize from 'lodash/memoize';
+import shallowEqual from 'react-pure-render/shallowEqual';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+
+/**
+ * Constants
+ */
+const VALID_ARG_TYPES = [ 'number', 'boolean', 'string' ];
+
+/**
+ * At runtime, assigns a function which returns a cache key for the memoized
+ * selector function, given a state object and a variable set of arguments. In
+ * development mode, this warns when the memoized selector is passed a complex
+ * object argument, as these cannot be depended upon as reliable cache keys.
+ *
+ * @type {Function} Function returning cache key for memoized selector
+ */
+const getCacheKey = ( () => {
+	if ( 'development' !== config( 'env' ) ) {
+		return ( state, ...args ) => args.join();
+	}
+
+	const warn = require( 'lib/warn' );
```

Comment:
huh. does this actually keep these out of the prod bundle? I thought the analyzer was static...